packages/lib/test/src/helpers/encodeCall.test.js

```diff
@@ -0,0 +1,192 @@
+    it('should accept valid values', () => {
+      assertGoodEncoding(['string'], ['hello']);
+      assertGoodEncoding(['string'], ['0x123']);
+      assertGoodEncoding(['string'], ['42']);
+      assertGoodEncoding(['string'], ['0x39af68cF04Abb0eF8f9d8191E1bD9c041E80e18e']);
+    });
+
+    it('should throw when the passed value is not a string nor a Buffer', () => {
+      assertBadEncoding(['string'], [2], /invalid string value/);
+      assertBadEncoding(['string'], [{}], /invalid string value/);
+    });
+  });
+  
+  describe('when the specified type is bytes (includes bytes1, bytes2, etc)', () => {
+    it('should accept valid values', () => {
+      assertGoodEncoding(['bytes'], ['0x2a']);
+      assertGoodEncoding(['bytes'], ['0xabcd']);
+      assertGoodEncoding(['bytes'], ['0xabcdef']);
+      assertGoodEncoding(['bytes'], [Buffer.from('')]);
+      assertGoodEncoding(['bytes'], [Buffer.from('hello')]);
+    });
+
+    it('should throw when a byte array expressed as a hexadecimal string is invalid', () => {
+      assertBadEncoding(['bytes'], ['0xabcqqq'], /invalid bytes value/);
+    });
+
+    it('should throw when the passed value is not a string nor a Buffer', () => {
+      assertBadEncoding(['bytes'], [2], /invalid bytes value/);
+      assertBadEncoding(['bytes'], [{}], /invalid bytes value/);
+    });
+  });
+  
+  describe('when the specified type is an address', () => {
+    it('should accept valid values', () => {
+      assertGoodEncoding(['address'], ['0x39af68cF04Abb0eF8f9d8191E1bD9c041E80e18e']);
+    });
+
+    it('should throw when an invalid address is passed', () => {
+      assertBadEncoding(['address'], ['0x39af68cF04Abb8f9d8191E1bD9ce18e'], /invalid address/);
+      assertBadEncoding(['address'], ['0x00000000000000000000000000000000000000000000000000000000f8a8fd6d'], /invalid address/);
+    });
+
+    it('should throw when an address with upper and lower case chars and an invalid checksum is passed', () => {
+      assertBadEncoding(['address'], ['0xCF5609B003b2776699eeA1233F7C82d5695CC9AA'], /invalid address/);
+    });
+
+    it('should not throw when an address with an invalid checksum is passed, if the address contains all upper or lower case strings', () => {
+      expect(() => encodeCall(NAME, ['address'], ['0xCF5609B003B2776699EEA1233F7C82D5695CC9AA'])).to.not.throw();
+      expect(() => encodeCall(NAME, ['address'], ['0xcf5609b003b2776699eea1233f7c82d5695cc9aa'])).to.not.throw();
+    });
+
+    it('should throw when the passed value is not a string nor a Buffer', () => {
+      assertBadEncoding(['address'], [2], /invalid address/);
+      assertBadEncoding(['address'], [{}], /invalid address/);
+    });
+  });
+  
+  // TODO: ethers.js/abi-coder does not support ufixed and fixed types?
+  // describe('when the specified type is a function', () => {});
+
+  // TODO: ethers.js/abi-coder does not support function types?
+  // describe('when the specified type is a function', () => {});
```

Comment:
move these TODOs to issues if necessary please

Reply:
I already did. I'll just delete the TODO comments. Thx!